lib/spring/application.rb

```diff
@@ -319,7 +319,10 @@ def shush_backtraces
             ensure
               if $!
                 lib = File.expand_path("..", __FILE__)
-                $!.backtrace.reject! { |line| line.start_with?(lib) } unless $!.backtrace.frozen?
+                unless $!.backtrace.frozen?
+                  $!.backtrace.reject! { |line| line.start_with?(lib) }
+                  $!.backtrace_locations.reject! { |line| line.path.start_with?(lib) }
+                end
               end
```

Comment:
How do we feel about giving `$!` a more reasonable name?

```suggestion
              error = $!
              if error
                lib = File.expand_path("..", __FILE__)
                unless error.backtrace.frozen?
                  error.backtrace.reject! { |line| line.start_with?(lib) }
                  error.backtrace_locations.reject! { |line| line.path.start_with?(lib) }
                end
              end
```